streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/bolt/query/WindowedQueryBolt.java

```diff
@@ -198,8 +195,12 @@ public void withWindowConfig(Window windowConfig) throws IOException {
             withLag(new BaseWindowedBolt.Duration(windowConfig.getLagMs(), TimeUnit.MILLISECONDS));
         }
 
-        if (windowConfig.getTsField() != null) {
-            withTimestampField(windowConfig.getTsField());
+        if (windowConfig.getTsFields() != null  &&  windowConfig.getTsFields().length>0) {
```

Comment:
What about the existing Join bolts having tsField in the config. How would that get handled ?

Reply:
Existing Join processor does not have tsField in the config from UI.

Reply:
You mean the condition `windowConfig.getTsField()` was always null earlier? If so its ok.

Reply:
right.